velox/experimental/cudf/exec/CudfHashJoin.cpp

```diff
@@ -147,7 +148,13 @@ void CudfHashJoinBuild::noMoreInput() {
   };
 
   auto stream = cudfGlobalStreamPool().get_stream();
-  auto tbl = getConcatenatedTable(inputs_, stream);
+  std::unique_ptr<cudf::table> tbl;
+  if (inputs_.size() == 0) {
+    auto emptyRowVector = RowVector::createEmpty(joinNode_->sources()[1]->outputType(), operatorCtx_->pool());
+    tbl = facebook::velox::cudf_velox::with_arrow::toCudfTable(emptyRowVector, operatorCtx_->pool(), stream);
+  } else {
+    tbl = getConcatenatedTable(inputs_, stream);
+  }
```

Comment:
If an input is empty, does it have type information? Could this empty table generation be done inside `getConcatenatedTable`?

Reply:
If the input size is zero, it will have type information. But if the input is never added (inputs_ is empty), it does not have any type information.